rero_ils/modules/libraries/tasks.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019-2023 RERO
+# Copyright (C) 2019-2023 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Tasks on `Library` resource."""
+
+from celery import shared_task
+from invenio_cache import current_cache
+
+
+@shared_task(bind=True, name='library-calendar-changes-update-loans')
+def calendar_changes_update_loans(self, record_data):
+    """Task used to update...
+
+    :param self: the task itself
+    :param record_data: Data representing the library to check.
+    """
+
+    def _at_finish():
+        """Inner method called when the task finished."""
+        # DEV NOTES :: Clean the cache to remove task_id when finished
+        #   We build a behavior ot avoid multiple concurrent similar tasks at
+        #   same time. The parent process running the task should place a stamp
+        #   when the task is registered. So, to specify another similar task
+        #   must be started without collision, we need to clean this stamp.
+        # DEV NOTES :: Why not using 'decorator'
+        #   A better way should be to use `@clean_cache` decorator ; this
+        #   decorator should take the key to clean as argument. But we didn't
+        #   know the key because it's created from `record_data`. This is why
+        #   it's easier to create a small specific function.
+        cache_content = current_cache.get('library-calendar-changes') or {}
+        cache_content.pop(library.pid, {})
+        current_cache.set('library-calendar-changes', cache_content)
+
+    from rero_ils.modules.loans.api import LoansIndexer, \
+        get_on_loan_loans_for_library
+    from .api import Library
+
+    library = Library(record_data)
+    active_loan_counter, changed_loan_uuids = 0, []
```

Comment:
Yes it is possible to write it like this, but for readability we can effort one line more ?
```python
active_loan_counter = 0
changed_loan_uuids = []
```